server/routes.go

```diff
@@ -1967,88 +1967,151 @@ func (s *Server) ChatHandler(c *gin.Context) {
 		toolParser = tools.NewParser(m.Template.Template, req.Tools)
 	}
 
+	structuredOutputsStarted := false
```

Comment:
I think this should be in the goroutine below since it's only written/read there?